concordia/admin/views.py

```diff
@@ -192,27 +208,30 @@ def admin_bulk_import_review(request):
                             continue
 
                         try:
+                            urls.append(url)
+                            url_counter = url_counter + 1
 
-                            count = event_loop.run_until_complete(
-                                import_item_count_from_url(url)
-                            )
-                            total_count = total_count + count
+                            if url_counter == 30:
+                                all_urls.append(urls)
+                                url_counter = 0
+                                urls = []
 
-                            messages.info(
-                                request,
-                                f"{project_title} - [{url}] : {count}",
-                            )
                         except Exception as exc:
                             messages.error(
                                 request,
                                 f"Unhandled error attempting to count {url}: {exc}",
                             )
+
+                all_urls.append(urls)
+                for i, val in enumerate(all_urls):
+                    return_result = fetch_all_urls(val)
+                    messages.info(request, f"Asset Count:{return_result[0]}")
+                    sum_count = sum_count + return_result[1]
+                    time.sleep(10)
```

Comment:
This will block the current process until the task returns. If the intention is to prevent overloading the remote server it might be better to add some delay to the actual request function to avoid issuing 15 requests at the same time and then idling for 10 seconds after the last one responds. A simple `time.sleep(random.random())` would avoid a dog-pile, and I especially like using that when retrying errors — note that this is extremely easy with something like [backoff](https://pypi.org/project/backoff/) on the actual retrieval function.